### DISCUSSION TITLE
Cyclic recursion when serialising object in table.add()

### DISCUSSION THREAD

**Discussion by datdinhquoc** [dexie/Dexie.js#1320] [Q&A] [answered]
Do people face this issue that I'm facing, it's cyclic recursion when inserting new doc.

I have these classes:
```
class Foo{
    constructor(){
        this.bar = new Bar(this);
    }
}

class Bar{
    constructor(Parent){
        this.Parent = Parent;
    }
}
```

When saving an instance of Foo using table.add(), an error about cyclic recursion is shown.

**Comment by dfahlander** [comment, marked as answer]
Try dexie@3.1.0-beta.11. It should resolve cyclic references correctly.
